pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -0,0 +1,204 @@
+				if err != nil {
+					log.Log.Reason(err).Error("Qemu agent poller error")
+					continue
+				}
+				interfaceStatuses := p.GetInterfaceStatuses(cmdResult)
+				if !reflect.DeepEqual(p.domainData.interfaces, interfaceStatuses) {
+					p.domainData.interfaces = interfaceStatuses
+
+					agentUpdateEvent := AgentUpdateEvent{
+						InterfaceStatuses: &interfaceStatuses,
+						DomainName:        p.domainData.name,
+					}
+					p.agentUpdateChan <- agentUpdateEvent
+				}
+			}
+		}
+	}()
+}
+
+func (p *AgentPoller) Stop() {
+	if p.agentDone != nil {
+		close(p.agentDone)
+		p.agentDone = nil
+	}
+}
+
+func (p *AgentPoller) GetInterfaceStatuses(cmdResult string) []api.InterfaceStatus {
+	parsedResult := parseAgentReply(cmdResult)
+	macToIp, macToIps, macToName := processAgentReply(parsedResult)
+	return p.createInterfaceStatuses(macToIp, macToIps, macToName)
+}
+
+func (p *AgentPoller) UpdateDomain(domain *api.Domain) {
+	if domain != nil {
+		select {
+		case p.domainUpdate <- domain:
+		default:
+			log.Log.Error("Failed to upate agent poller domain info")
+		}
+	}
+}
+
+func (p *AgentPoller) createDomainData(domain *api.Domain) DomainData {
+	aliases := []string{}
+	aliasToMac := map[string]string{}
+	for _, ifc := range domain.Spec.Devices.Interfaces {
+		mac := ifc.MAC.MAC
+		alias := ifc.Alias.Name
+		aliasToMac[alias] = mac
+		aliases = append(aliases, alias)
+	}
+	return DomainData{
+		name:       domain.Spec.Name,
+		aliases:    aliases,
+		aliasToMac: aliasToMac,
+	}
+}
+
+func (p *AgentPoller) pollQemuAgent(domainName string) (string, error) {
+	cmdResult, err := p.Connection.QemuAgentCommand("{\"execute\":\"guest-network-get-interfaces\"}", domainName)
+	return cmdResult, err
+}
+
+func parseAgentReply(agentReply string) Result {
+	result := Result{}
+	json.Unmarshal([]byte(agentReply), &result)
+	return result
+}
+
+func (p *AgentPoller) createInterfaceStatuses(macToIp map[string]string, macToIps map[string][]string, macToName map[string]string) []api.InterfaceStatus {
+	interfaceStatuses := []api.InterfaceStatus{}
+	aliasedMacs := []string{}
+	for _, alias := range p.domainData.aliases {
+		mac := p.domainData.aliasToMac[alias]
+		aliasedMacs = append(aliasedMacs, mac)
+		ifcStatus := api.InterfaceStatus{
+			Name:          alias,
+			Mac:           mac,
+			Ip:            macToIp[mac],
+			IPs:           macToIps[mac],
+			InterfaceName: macToName[mac],
+		}
+		interfaceStatuses = append(interfaceStatuses, ifcStatus)
+		delete(macToIp, mac)
+	}
+	for mac, ip := range macToIp {
+		if macToName[mac] == "lo" {
+			continue
+		}
+		ifcStatus := api.InterfaceStatus{
+			Mac:           mac,
+			Ip:            ip,
+			IPs:           macToIps[mac],
+			InterfaceName: macToName[mac],
+		}
+		interfaceStatuses = append(interfaceStatuses, ifcStatus)
+
+	}
+	return interfaceStatuses
+}
+
+func processAgentReply(agentResult Result) (map[string]string, map[string][]string, map[string]string) {
```

Comment:
could you please add a comments describing what is being processed and what is the goal here?

Reply:
Refactored the code. Should be more readable now.